File: tests/python/cases/case_238_helper_sink_dispatch_04.py
"""
Task case 238: helper_sink_dispatch_04 deep stress pattern.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_238_helper_sink_dispatch_04(input_data):
    task_name = 'helper_sink_dispatch_04'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'extreme',
        'stage': 'stress',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    def sink(obj):
        RETAINED_AUDIT.append(obj)
    # ESCAPE: helper function dispatch hides sink call site.
    sink(payload)
    return 'ok'

File: tests/nodejs/cases/case_238_helper_sink_dispatch_04.js
/**
 * Task case 238: helper_sink_dispatch_04 deep stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case238HelperSinkDispatch04(input) {
  const taskName = 'helper_sink_dispatch_04';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'extreme',
    stage: 'stress',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const sink = (obj) => retainedCache.set('helper', obj);
  // ESCAPE: helper function obscures sink dispatch.
  sink(payload);
  return 'ok';
}

module.exports = { case238HelperSinkDispatch04 };
